tests/performance/shatal/internal/shatal/modifier.go

```diff
@@ -0,0 +1,81 @@
+package shatal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/deckhouse/virtualization/api/core/v1alpha2"
+
+	"github.com/deckhouse/virtualization/shatal/internal/api"
+)
+
+// Modifier updates virtual machines (core fraction from 10% to 25% and vice versa).
+type Modifier struct {
+	api       *api.Client
+	namespace string
+	logger    *slog.Logger
+}
+
+func NewModifier(api *api.Client, namespace string, log *slog.Logger) *Modifier {
+	return &Modifier{
+		api:       api,
+		namespace: namespace,
+		logger:    log.With("type", "modifier"),
+	}
+}
+
+func (s *Modifier) Do(ctx context.Context, vm v1alpha2.VirtualMachine) {
+	if vm.Spec.CPU.CoreFraction == "10%" {
+		vm.Spec.CPU.CoreFraction = "25%"
+	} else {
+		vm.Spec.CPU.CoreFraction = "10%"
+	}
+
+	s.logger.
+		With("node", vm.Status.NodeName).
+		With("core-fraction", vm.Spec.CPU.CoreFraction).
+		Info(fmt.Sprintf("Modify: %s", vm.Name))
+
+	err := s.api.UpdateVM(ctx, vm)
```

Comment:
I have a feeling that Patch is better, as Update may lead to error "unable to update ... the object has been modified; please apply your changes to the latest version and try again".